overlord/state/state.go

```diff
@@ -298,7 +299,7 @@ func (s *State) Cached(key interface{}) interface{} {
 // The cached value is not persisted.
 func (s *State) Cache(key, value interface{}) {
 	s.reading() // Doesn't touch persisted data.
-	if value == nil {
+	if value == nil || (reflect.ValueOf(value).Kind() == reflect.Map && len(reflect.ValueOf(value).MapKeys()) == 0) {
```

Comment:
we don't want this change like this, this is replacing an empty map with deleting it which might create a different set of bugs

Reply:
Mmm... good point...